packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -205,24 +205,29 @@ function transform(codeGen: CodeGen): t.Expression {
      * Transforms an IfBlock or ElseifBlock along with both its direct descendants and its 'else' descendants.
      *
      * @param conditionalParentBlock The IfBlock or ElseifBlock to transform into a conditional expression
+     * @param key The key to use for this chain of IfBlock/ElseifBlock branches, if applicable
      * @returns A conditional expression representing the full conditional tree with conditionalParentBlock as the root node
      */
     function transformConditionalParentBlock(
-        conditionalParentBlock: IfBlock | ElseifBlock
+        conditionalParentBlock: IfBlock | ElseifBlock,
+        key?: string | number
     ): t.Expression {
+        const ifBlockKey = key || codeGen.generateKey('if-fr');
```

Comment:
We use the same key for each branch of the if-elseif-else chain so that the different fragment vnodes are recognized as belonging to the same space. Thanks Jose for the recommendation.